controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -499,15 +516,47 @@ func (r *StorageClassClaimReconciler) reconcileProviderPhases() (reconcile.Resul
 	if err := r.get(r.storageConsumer); err != nil {
 		return reconcile.Result{}, err
 	}
-
+	profile := ""
+	// check claim status already contains the name of the resource. if not, add it.
 	if r.storageClassClaim.Spec.Type == "blockpool" {
 		r.cephBlockPool = &rookCephv1.CephBlockPool{}
-		r.cephBlockPool.Name = fmt.Sprintf("cephblockpool-%s", r.storageConsumer.Name)
 		r.cephBlockPool.Namespace = r.OperatorNamespace
+		for _, res := range r.storageClassClaim.Status.CephResources {
+			if res.Kind == "CephBlockPool" {
+				r.cephBlockPool.Name = res.Name
+				profile = r.storageClassClaim.Status.StorageProfile
+				break
+			}
+		}
+		if r.cephBlockPool.Name == "" {
+			r.cephBlockPool.Name = fmt.Sprintf("cephblockpool-%s-%s", r.storageConsumer.Name, generateUUID())
+			if r.storageClassClaim.Spec.StorageProfile == "" {
+				profile = r.storageCluster.Spec.DefaultStorageProfile
+			}
+
+		}
+
 	} else if r.storageClassClaim.Spec.Type == "sharedfilesystem" {
 		r.cephFilesystemSubVolumeGroup = &rookCephv1.CephFilesystemSubVolumeGroup{}
-		r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("cephfilesystemsubvolumegroup-%s", r.storageConsumer.Name)
 		r.cephFilesystemSubVolumeGroup.Namespace = r.OperatorNamespace
+		for _, res := range r.storageClassClaim.Status.CephResources {
+			if res.Kind == "CephFilesystemSubVolumeGroup" {
+				r.cephFilesystemSubVolumeGroup.Name = res.Name
+				profile = r.storageClassClaim.Status.StorageProfile
+				break
+			}
+		}
+		if r.cephFilesystemSubVolumeGroup.Name == "" {
+			r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("cephfilesystemsubvolumegroup-%s-%s", r.storageConsumer.Name, generateUUID())
+			if r.storageClassClaim.Spec.StorageProfile == "" {
+				profile = r.storageCluster.Spec.DefaultStorageProfile
+			}
+		}
+	}
```

Comment:
This will be problematic for the upgraded cluster it will change the default storage profile, which inturns can choose different deviceClass for existing cephfs/rbd resources.